### PR TITLE
[INF] Devcontainer

### PR DESCRIPTION
**Pull request by ericmjl**
This PR will give us a development container.

README is on its way to be updated, with instructions on how to get setup using VSCode.

@fonnesbeck and @michaelosthege have test-driven it out. Others please test-drive it too!

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
```
Comment (michaelosthege) [inline .devcontainer/Dockerfile:4]: You might want to change that copyright notice?

File: README.md [hunk 2]
```diff
@@ -18,3 +18,27 @@ However, expect things to break or change without warning.
 
 See here for an example: https://github.com/pymc-devs/pymc4/blob/master/notebooks/radon_hierarchical.ipynb
 See here for the design document: https://github.com/pymc-devs/pymc4/blob/master/notebooks/pymc4_design_guide.ipynb
+
+## Develop
+
+One easy way of developing on PyMC4 is to take advantage of the development containers! 
+Using pre-built development environments allows you to develop on PyMC4 without needing to set up locally.
+
+To use the dev containers, you will need to have Docker and VSCode running locally on your machine, and will need the VSCode Remote extension installed inside VSCode.
```
Comment (michaelosthege) [inline README.md:27]: Can you be more specific about the VSCode Remote extension? (Which version, or link to marketplace..)

File: README.md [hunk 3]
```diff
@@ -18,3 +18,27 @@ However, expect things to break or change without warning.
...
+2. Create a new branch inside your fork
+3. Copy the branch URL
+
+Now, in VSCode:
+
+1. In the command palette, search for "Remote-Containers: Open Repository in Container...".
+2. Paste in the branch URL
+3. If prompted, create it in a "Unique Volume".
+
+Happy hacking away! Don't forget to commit your changes and push them to your branch.
```
Comment (michaelosthege) [inline README.md:41]: If the changes are really lost when VSCode is closed, the "Don't forget to.." warning should be more prominent

File: README.md (hunk 2 above)
Comment (ericmjl) [reply README.md:27]: Good point! Let me modify.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (ericmjl) [reply .devcontainer/Dockerfile:4]: Hmmm, it is modified from Microsoft's suggested Dockerfile though, and I think it has to be propagated, as per the license in the URL. Specifically:

>     The above copyright notice and this permission notice shall be included in all
>     copies or substantial portions of the Software.

File: README.md (hunk 3 above)
Comment (ericmjl) [reply README.md]: Done!

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// For format details, see https://aka.ms/vscode-remote/devcontainer.json or this file's README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.117.1/containers/python-3-miniconda
+{
+	"name": "PyMC4 Development Container",
+	"context": "..",
+	"image": "registry.hub.docker.com/ericmjl/pymc4:devcontainer",
```
Comment (ericmjl) [inline .devcontainer/devcontainer.json:6]: Btw, someone owns the pymc Docker repository namespace, so I don't have privileges to set this up. @twiecki or @fonnesbeck, do any of you know who is in charge of it?

Comment (ericmjl) [reply .devcontainer/devcontainer.json:6]: I've got the necessary permissions.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (michaelosthege) [reply .devcontainer/Dockerfile:4]: We then keep it I guess, but maybe link to the original?

Comment (ericmjl) [reply .devcontainer/Dockerfile:5]: Done!

**Pull request merged**